### DISCUSSION TITLE
v1.7.0-alpha1

### DISCUSSION THREAD

**Discussion by ghost** [opentofu/opentofu#1402] [Announcements]
⚠️ Do not use this release for production workloads! ⚠️

It's time for the [first prerelease of the 1.7.0 version](https://github.com/opentofu/opentofu/releases/tag/v1.7.0-alpha1)! This includes a lot of major and minor new features, as well as a ton of community contributions!

The highlights are:
- State Encryption ([docs](https://1-7-0-alpha1.opentofu.pages.dev/docs/language/state/encryption/))
- Declarative `removed` blocks ([docs](https://1-7-0-alpha1.opentofu.pages.dev/docs/language/resources/syntax/#removing-resources))

[Please take it for a spin and let us know your feedback!](https://github.com/opentofu/opentofu/releases/tag/v1.7.0-alpha1)

For all the features, see the [detailed changelog](https://github.com/opentofu/opentofu/blob/8c99c7522925b9c9a6cebe8e22138fa39e9706ed/CHANGELOG.md#170-unreleased).

You can find the full diff [here](https://github.com/opentofu/opentofu/compare/v1.6...v1.7.0-alpha1?expand=1).

**Comment by ghost** [comment]
Hey folks, please help us test this release. The detailed instructions are available [in this blog post](https://opentofu.org/blog/help-us-test-opentofu-1-7-0-alpha1/).

**Comment by dylanmtaylor** [comment]
There are allegations that the removed block code was stolen from BUSL licensed hashicorp source code. Please look into this and revert the commit if this is indeed the case. This feature needs to be clean room engineered.

**Reply by marcinwyszynski** [reply]
@dylanmtaylor OpenTofu vehemently disagrees with any suggestion that it misappropriated, mis-sourced, or otherwise misused HashiCorp’s BSL code. Indeed, it seems that HashiCorp may be conflating code that it had previously been open-sourced under the MPL and more recently developed code it published under the BSL. OpenTofu’s maintainers have investigated this matter, and intend to issue a written response providing a more detailed explanation of its position in the coming days.
